src/sinks/aws_s3.rs

```diff
@@ -12,7 +14,10 @@ use crate::{
 use bytes::Bytes;
 use chrono::Utc;
 use futures::{stream::iter_ok, Future, Poll, Sink};
+use rusoto_core::HttpClient;
 use rusoto_core::{Region, RusotoError, RusotoFuture};
+#[cfg(not(test))]
+use rusoto_credential::DefaultCredentialsProvider;
```

Comment:
minor nit: can we move this use down into the block that already has `cfg(not(test))` applied to it like how we do it for staticprovider.

Reply:
Yes, that's better